### PR TITLE
Update license from Apache to MIT

### LINKED ISSUE #18873: Switch to MIT license
See also https://github.com/dotnet/roslyn/issues/37797

### PR DESCRIPTION
**Pull request by Pilchie**
Fixes #18873.

Most work done by either Find and Replace or the Visual Studio fix to add missing headers.

### CODE REVIEW

**Comment by Pilchie** [comment]
I suggest reviewing commit by commit, and skipping the second and last commits as they are fairly mechanical.

**Comment by Pilchie** [comment]
@sharwell @richlander @ChrisSfanos

File: src/Components/test/E2ETest/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "sauce": "ts-node ./scripts/sauce.ts"
   },
   "author": "",
-  "license": "Apache-2.0",
```
Comment (Pilchie) [inline src/Components/test/E2ETest/package.json:13]: There are a bunch of `package.json` files with author of empty or Microsoft, should those be updated?

File: src/Installers/Debian/Runtime/debian_config.json.in [hunk 2]
```diff
@@ -22,10 +22,10 @@
     "architecture":"any"
   },
 
-  "copyright": "Microsoft",
+  "copyright": ".NET Foundation and Contributors",
```
Comment (Pilchie) [inline src/Installers/Debian/Runtime/debian_config.json.in:25]: Is this the right thing to do?  @dleeapho ?

File: src/Mvc/Settings.StyleCop [hunk 3]
```diff
@@ -439,12 +439,6 @@
       </Rules>
       <AnalyzerSettings />
     </Analyzer>
-    <Analyzer AnalyzerId="StyleCop.KRules.FileHeaderRule">
-      <Rules />
-      <AnalyzerSettings>
-        <StringProperty Name="FileHeaderText"> Copyright (c) Microsoft Open Technologies, Inc. All rights reserved. Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.</StringProperty>
```
Comment (Pilchie) [inline src/Mvc/Settings.StyleCop]: Figured this was no longer relevant

**Comment by BrennanConroy** [comment]
Are we going to handle non cs files later?

**Comment by Pilchie** [comment]
> Are we going to handle non cs files later?

The Find and Replace step got them.  I did a `git grep Apache` and walked through the results, and AFAICT, everything remaining after this is redists of other cases (like jquery).

**Comment by BrennanConroy** [comment]
Oh, probably the second commit, which we can't really look at :D

**Comment by Pilchie** [comment]
Yes, that's my expectation.

**Comment by Pilchie** [comment]
`git show 3dbb4f19f6 --name-only  | findstr .java | measure-object` confirms it (and other extensions too)

**Comment by BrennanConroy** [comment]
Looks like this wasn't done properly:
https://github.com/dotnet/aspnetcore/blob/e954b6aef98408a7b672c7ce31e8dad0b24345f8/src/SignalR/clients/java/signalr/build.gradle#L34-L36

**Review by pranavkm — APPROVED** [review]

**Comment by pranavkm** [comment]
#yolo approval. I can't quite review all the files, but the general gist of it seems right. 

@Pilchie I would recommend starting this as a suggestion and doing a second pass that turns in to Warnings and clearing up the build warnings. Less chances that you run in to perpetual file conflicts that way.

**Comment by Pilchie** [comment]
@dotnet/aspnet-build Looks like some provisioning failures here.  What's the best way to re-run?

**Comment by Pilchie** [comment]
It's green - going to merge!

**Pull request merged**